hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/SnapshotDefragService.java

```diff
@@ -0,0 +1,1138 @@
+//      try (ManagedCompactRangeOptions compactOptions = new ManagedCompactRangeOptions()) {
+//        compactOptions.setChangeLevel(true);
+//        compactOptions.setTargetLevel(1);
+//        currentDefraggedDb.compactDB(compactOptions);
+//      }
+//      LOG.info("Completed compaction of incrementally defragmented DB");
+
+      // 5. Verify data integrity
+      verifyDbIntegrity(currentSnapshotDb, currentDefraggedDb, currentSnapshot);
+
+      // Close the defragmented DB. TODO: Close in finally block instead
+      currentDefraggedStore.close();
+
+      LOG.info("Successfully completed incremental defragmentation for snapshot: {} with {} incremental changes",
+          currentSnapshot.getName(), incrementalKeysCopied);
+
+    } catch (RocksDatabaseException e) {
+      LOG.error("RocksDB error during incremental defragmentation of snapshot: {}",
+          currentSnapshot.getName(), e);
+//      LOG.warn("Falling back to full defragmentation due to error");
+//      performFullDefragmentation(currentSnapshot, currentOmSnapshot);
+    } catch (Exception e) {
+      LOG.error("Unexpected error during incremental defragmentation of snapshot: {}",
+          currentSnapshot.getName(), e);
+      LOG.warn("Falling back to full defragmentation due to error");
+      performFullDefragmentation(currentSnapshot, currentOmSnapshot);
+    }
+  }
+
+  /**
+   * Applies incremental changes by using snapshotDiff to compute the diff list,
+   * then iterating that diff list against the current snapshot checkpoint DB.
+   * Uses RDBSstFileWriter to directly write changes to SST files and then ingests them.
+   */
+  @SuppressWarnings("checkstyle:MethodLength")
+  private long applyIncrementalChanges(RocksDatabase currentSnapshotDb, DBStore targetStore,
+      SnapshotInfo currentSnapshot, SnapshotInfo previousSnapshot) throws RocksDatabaseException {
+
+    LOG.info("Applying incremental changes for snapshot: {} since: {} using snapshotDiff approach",
+        currentSnapshot.getName(), previousSnapshot.getName());
+
+    long totalChanges = 0;
+
+    // Create temporary directory for SST files
+    String currentSnapshotPath = OmSnapshotManager.getSnapshotPath(
+        ozoneManager.getConfiguration(), currentSnapshot);
+    String parentDir = Paths.get(currentSnapshotPath).getParent().getParent().toString();
+    String tempSstDir = Paths.get(parentDir, CHECKPOINT_STATE_DEFRAGGED_DIR, TEMP_DIFF_DIR).toString();
+
+    try {
+      Files.createDirectories(Paths.get(tempSstDir));
+      LOG.info("Created temporary SST directory: {}", tempSstDir);
+
+      // Use snapshotDiff to compute the diff list between previous and current snapshot
+      LOG.info("Computing snapshot diff between {} and {}",
+          previousSnapshot.getName(), currentSnapshot.getName());
+
+      SnapshotDiffResponse diffResponse;
+      try {
+        // Call snapshotDiff to get the diff list
+        diffResponse = ozoneManager.snapshotDiff(
```

Comment:
Instead of calling snapshot diff we can directly use the getDeltaFilesFunction and 
https://github.com/apache/ozone/blob/13ccecaea1a166010175e68a4d3c9a641764d141/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/SnapshotDiffManager.java#L1171
and use 
`Stream<String> keysToCheck = new SstFileSetReader(deltaFiles).getKeyStreamWithTombstone();`
Now for each keysToCheck previousSnapshot.get(keyToCheck) != currentSnapshot.get(keyToCheck) byte level comparison should be good enough here no need to deserialize and put to sstFileWriter only if the byte arrays are not equal.

Reply:
done